cmd/openshift-install/agent.go

```diff
@@ -15,7 +20,67 @@ func newAgentCmd() *cobra.Command {
 		},
 	}
 
-	agentCmd.AddCommand(agent.NewCreateCmd())
+	agentCmd.AddCommand(newAgentCreateCmd())
 	agentCmd.AddCommand(agent.NewWaitForCmd())
 	return agentCmd
 }
+
+var (
+	agentManifestsTarget = target{
+		name: "Manifests",
+		command: &cobra.Command{
+			Use:   "manifests",
+			Short: "Generates intermediate files required by the agent installer",
+			Args:  cobra.ExactArgs(0),
+		},
+		assets: []asset.WritableAsset{
+			&manifests.AgentManifests{},
+		},
+	}
+
+	agentImageTarget = target{
+		name: "Image",
+		command: &cobra.Command{
+			Use:   "image",
+			Short: "Generates a bootable image containing all the information needed to deploy a cluster",
+			Args:  cobra.ExactArgs(0),
+			PostRun: func(cmd *cobra.Command, args []string) {
+				cleanup := setupFileHook(rootOpts.dir)
+				defer cleanup()
+
+				timer.StartTimer("Agent Create Image Complete")
+
+				err := agentcmd.BuildImage()
+				if err != nil {
+					logrus.Fatal(err)
+				}
+
+				timer.StopTimer("Agent Create Image Complete")
+				timer.LogSummary()
+			},
+		},
+		assets: nil,
+	}
+
+	agentTargets = []target{agentManifestsTarget, agentImageTarget}
+)
+
+func newAgentCreateCmd() *cobra.Command {
+
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Commands for generating agent installer based artifacts",
```

Comment:
```suggestion
		Short: "Commands for generating agent installation artifacts",
```
This might be more accurate.